enterprise/frontend/src/metabase-enterprise/auth/components/SettingsJWTForm/SettingsJWTForm.jsx

```diff
@@ -1,19 +1,57 @@
-import { useCallback } from "react";
+import { useCallback, useMemo } from "react";
 import PropTypes from "prop-types";
 import { connect } from "react-redux";
 import { t } from "ttag";
+import _ from "underscore";
 import { updateSettings } from "metabase/admin/settings/settings";
-import SettingsBatchForm from "metabase/admin/settings/components/SettingsBatchForm";
-import { FormButton } from "./SettingsJWTForm.styled";
+import {
+  FormErrorMessage,
+  FormProvider,
+  FormSecretKey,
+  FormSubmitButton,
+  FormTextInput,
+} from "metabase/forms";
+import Breadcrumbs from "metabase/components/Breadcrumbs";
+import { Stack } from "metabase/ui";
+import { FormSection } from "metabase/containers/FormikForm";
+import GroupMappingsWidget from "metabase/admin/settings/containers/GroupMappingsWidget";
+import { JWTForm, JWTFormFooter } from "./SettingsJWTForm.styled";
 
 const propTypes = {
+  elements: PropTypes.array,
   settingValues: PropTypes.object.isRequired,
   onSubmit: PropTypes.func.isRequired,
 };
 
-const SettingsJWTForm = ({ settingValues, onSubmit, ...props }) => {
+const SettingsJWTForm = ({
+  elements = [],
+  settingValues,
+  onSubmit,
+  ...props
+}) => {
   const isEnabled = settingValues["jwt-enabled"];
 
+  const settings = useMemo(() => {
+    return _.indexBy(elements, "key");
+  }, [elements]);
+
+  const fields = useMemo(() => {
+    return _.mapObject(settings, setting => ({
```

Comment:
[`Object.values`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Object/values) will only get you the values. In this case there would be a bit more legwork to use native API.

It would look something like
```ts
Object.fromEntries(Object.entries(settings).map(([key, value]) => mapYourObjectHere))
```

but I wouldn't mind `mapObject` here either we seem to be using underscore everywhere, so I won't object using them if needed.